kafka-connect-s3/src/main/java/io/confluent/connect/s3/util/S3ErrorUtils.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2021 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.connect.s3.util;
+
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.AmazonWebServiceRequest;
+import com.amazonaws.retry.PredefinedRetryPolicies;
+import io.confluent.connect.storage.common.util.StringUtils;
+import org.apache.kafka.connect.errors.ConnectException;
+import org.apache.kafka.connect.errors.RetriableException;
+
+import java.io.IOException;
+
+/**
+ * Utilities related to basic S3 error/exception analysis.
+ */
+public class S3ErrorUtils {
+
+  /**
+   * Return whether the given exception is a "retryable" exception.
+   * @param exception The exception to analyze
+   * @return true if the exception is retryable
+   */
+  private static boolean isRetryableException(Throwable exception) {
+    if (exception == null) {
+      return false;
+    }
+    if (exception instanceof IOException) {
+      if (exception.equals(exception.getCause())) {
```

Comment:
Instead of recursion should we use [ExceptionUtils.html#getRootCause](https://commons.apache.org/proper/commons-lang/javadocs/api-3.4/org/apache/commons/lang3/exception/ExceptionUtils.html#getRootCause(java.lang.Throwable))  as the current code may not handle case where its E1->E2->E3->E1

Reply:
Fixed.